browser/ui/views/brave_help_bubble/brave_help_bubble_host_view.cc

```diff
@@ -0,0 +1,217 @@
+                                        kColors,
+                                        kPositions,
+                                        std::size(kColors),
+                                        SkTileMode::kClamp);
+
+void SchedulePulsingAnimation(ui::Layer* layer) {
+  DCHECK(layer);
+
+  constexpr base::TimeDelta kPulsingDuration = base::Milliseconds(1000);
+
+  const gfx::Rect local_bounds(layer->bounds().size());
+  views::AnimationBuilder()
+      .SetPreemptionStrategy(
+          ui::LayerAnimator::IMMEDIATELY_ANIMATE_TO_NEW_TARGET)
+      .Repeatedly()
+      .SetDuration(kPulsingDuration)
+      .SetTransform(layer,
+                    gfx::GetScaleTransform(local_bounds.CenterPoint(), 0.7f),
+                    gfx::Tween::EASE_IN)
+      .At(kPulsingDuration)
+      .SetDuration(kPulsingDuration)
+      .SetTransform(layer,
+                    gfx::GetScaleTransform(local_bounds.CenterPoint(), 1.0f),
+                    gfx::Tween::EASE_OUT);
+}
+}  // namespace
+
+BraveHelpBubbleHostView::BraveHelpBubbleHostView() {
+  // Disable event handling to interact with the underlying element.
+  SetCanProcessEventsWithinSubtree(false);
+  SetPaintToLayer();
+  layer()->SetFillsBoundsOpaquely(false);
+  SetSize({kWidth, kHeight});
+}
+
+BraveHelpBubbleHostView::~BraveHelpBubbleHostView() = default;
+
+bool BraveHelpBubbleHostView::Show() {
+  if (help_bubble_) {
+    return false;
+  }
+
+  CHECK(tracked_element_ && !text_.empty());
+
+  auto* brave_help_bubble_delegate_view =
+      new BraveHelpBubbleDelegateView(this, text_);
+  help_bubble_ = views::BubbleDialogDelegateView::CreateBubble(
+      brave_help_bubble_delegate_view);
+  auto* frame_view = brave_help_bubble_delegate_view->GetBubbleFrameView();
+  frame_view->SetDisplayVisibleArrow(true);
+  bubble_widget_observation_.Observe(help_bubble_);
+
+  // Observes tracked element and host widget(browser frame) to know this host
+  // view's position update timing.
+  tracked_view_observation_.Observe(tracked_element_);
+  host_widget_observation_.Observe(GetWidget());
+
+  // To locate help bubble at more higher than other normal widget.
+  help_bubble_->SetZOrderLevel(ui::ZOrderLevel::kFloatingUIElement);
+
+  ui::ElementIdentifier id =
+      tracked_element_->GetProperty(views::kElementIdentifierKey);
+  CHECK(id);
+  // Listen activation state to to hide bubble.(ex, hide bubble when button
+  // clicked)
+  activated_subscription_ =
+      ui::ElementTracker::GetElementTracker()->AddElementActivatedCallback(
+          id, views::ElementTrackerViews::GetContextForView(tracked_element_),
+          base::BindRepeating(
+              &BraveHelpBubbleHostView::OnTrackedElementActivated,
+              weak_factory_.GetWeakPtr()));
+
+  // With this inactive launching, bubble will be hidden after activated.
+  help_bubble_->ShowInactive();
+
+  UpdatePosition();
+  SetVisible(true);
+
+  if (!gfx::Animation::PrefersReducedMotion()) {
```

Comment:
totally optional: When I discussed difference between PrefersReducedMotion() and ShouldRenderRichAnimation(), they said PrefersReduceMotion() is literally focusing on the animation that moves. So I guess ShouldRenderRichAnimation() goes better for pulsing animation?

Reply:
Thanks for the advice! Updated.

Reply:
I appreciate how Chromium has different flags for this. However, the term 'move' is somewhat vague. What exactly do you consider as 'moving'? It can't just be about direction, which is covered by 'translate.' The pulse animation involves scaling, which can also be seen as a form of motion.

This [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) document in the second paragraph states: 

> Such animations can trigger discomfort for those with [vestibular motion disorders](https://www.a11yproject.com/posts/understanding-vestibular-disorders/). Animations such as scaling or panning large objects can be vestibular motion triggers.

I think the object we're animating isn't large, so maybe we're OK. The ink drop animation in Chromium obeys the ShouldRenderRichAnimation flag, which involves scaling but the object is small.